pkg/api/handlers.go

```diff
@@ -229,6 +232,13 @@ func NewQuerierHandler(
 		nil,
 	)
 
+	queryapi := qapi.NewAPI(
+		engine,
+		querier.NewErrorTranslateSampleAndChunkQueryable(queryable), // Translate errors to errors expected by API.
+		logger,
+		nil,
+	)
```

Comment:
Can we put all the new handler under a feature flag?
Like if I want to use the previous way to communicate btw querier and qfe I should still be able to do that

Reply:
When you say a feature flag you mean a config that enables or disables the new communication?